### ISSUE TITLE
Reproducing np.convolve with astropy.convolution.convolve

### ISSUE THREAD

**Issue by mirca** [astropy/astropy#6264]
This is more like a question than a issue or bug report.

I wonder whether or not ``astropy.convolution.convolve`` is able to reproduce the behavior of ``numpy.convolve`` when ``mode='same'``.

See:

```
>>> from astropy.convolution import convolve
>>> import numpy as np
>>> x = np.array([1, 2, 3])
>>> y = np.array([5, 4, 3, 2, 1])
>>> np.convolve(y, x, mode='same')
array([14, 26, 20, 14,  8])
>>> convolve(y, x, normalize_kernel=False)
array([ 22.,  22.,  16.,  10.,   4.])
>>> convolve(y, x, normalize_kernel=True)
array([ 3.66666667,  3.66666667,  2.66666667,  1.66666667,  0.66666667])
>>> convolve(y, x, normalize_kernel=False, boundary=None)
array([  0.,  22.,  16.,  10.,   0.])
>>> convolve(y, x, normalize_kernel=False, boundary='fill')
array([ 22.,  22.,  16.,  10.,   4.])
>>> convolve(y, x, normalize_kernel=False, boundary='wrap')
array([ 23.,  22.,  16.,  10.,  19.])
>>> convolve(y, x, normalize_kernel=False, boundary='extend')
array([ 27.,  22.,  16.,  10.,   7.])
```

cc @keflavich

**Comment by mirca** [comment]
Also,

```
>>> convolve_fft(y, x, normalize_kernel=False)
array([ 14.,  26.,  20.,  14.,   8.])
>>> convolve(y, x, normalize_kernel=False)
array([ 22.,  22.,  16.,  10.,   4.])
```

**Comment by astrofrog** [comment]
@mirca - are you using the latest developer version of Astropy?

Also pinging @keflavich as the convolution expert

**Comment by mirca** [comment]
@astrofrog 
```
>>> astropy.__version__
'3.0.dev19236'
```

**Comment by keflavich** [comment]
Oh boy... this is a ridiculously big bug that was introduced in #5782 because we never tested with asymmetric kernels.  `convolve` implements the wrong definition of convolution now: the kernel is not reversed, but it should be!  I will work on a fix.

**Comment by keflavich** [comment]
@astrofrog any tips on the fastest way to clean and rebuild *only* the cython modules from convolution/?  I tried deleting the .so and .c files and found that deleting the .c files led to a build failure.

**Comment by keflavich** [comment]
I was wrong, this has been around possibly forever!

**Comment by astrofrog** [comment]
@keflavich - not sure, but I use ccache so the compilation of actual C code is almost instantaneous, and most of the time is spent on Cython.

**Comment by keflavich** [comment]
ok, no worries.  I have a fix in prep, and this *definitely* needs to be a 2.0 blocker... I can't believe we missed it for so long.

**Comment by astrofrog** [comment]
@mirca @keflavich - is this now fixed?

**Comment by keflavich** [comment]
Yes, closed by #6267.
